File: billing-domain/src/main/java/com/example/billing/domain/invoice/IllegalInvoiceTransitionException.java
package com.example.billing.domain.invoice;

public class IllegalInvoiceTransitionException extends RuntimeException {
    public IllegalInvoiceTransitionException(InvoiceStatus from, InvoiceStatus to) {
        super("Invoice 상태 전이 불가: " + from + " → " + to);
    }
}

File: billing-domain/src/main/kotlin/com/example/billing/domain/invoice/IllegalInvoiceTransitionException.kt
package com.example.billing.domain.invoice

/**
 * Invoice 상태 전이 불가 예외. 도메인이 직접 발행하므로 data class 가 아닌 일반 class.
 */
class IllegalInvoiceTransitionException(
    from: InvoiceStatus,
    to: InvoiceStatus,
) : RuntimeException("Invoice 상태 전이 불가: $from → $to")
